spec/controllers/api/v1/tasks_controller_spec.rb

```diff
@@ -0,0 +1,203 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::TasksController, type: :controller do
+  let(:user) { create(:user, :with_project) }
+  let(:project) { user.projects.first }
+  let(:params) {}
+
+  before { @request.headers[I18n.t('auth')] = JsonWebToken.encode(user_id: user.id) }
+
+  describe 'GET index' do
+    context 'when Success' do
+      context 'with valid project_id' do
+        let(:params) { { project_id: project.id } }
+
+        before do
+          create(:task, project: project)
+          get :index, params: params
+        end
+
+        it { expect(response).to have_http_status(:ok) }
+        it { expect(response).to match_response_schema('tasks') }
+      end
+    end
+
+    context 'when Failure' do
+      context 'with invalid project_id' do
+        let(:params) { { project_id: 0 } }
+
+        before { get :index, params: params }
+
+        it { expect(response).to have_http_status(:not_found) }
+        it { expect(JSON.parse(response.body)['errors']).to include(I18n.t('errors.project_not_found')) }
+      end
+    end
+  end
+
+  describe 'GET show' do
+    before { get :show, params: params }
+
+    context 'when Success' do
+      context 'with valid id' do
+        let(:task) { create(:task, project: project) }
+        let(:params) { { id: task.id } }
+
+        it { expect(response).to have_http_status(:ok) }
+        it { expect(response).to match_response_schema('task') }
+      end
+    end
+
+    context 'when Failure' do
+      context 'with invalid id' do
+        let(:params) { { id: 0 } }
+
+        it { expect(response).to have_http_status(:not_found) }
+        it { expect(JSON.parse(response.body)['errors']).to include(I18n.t('errors.task_not_found')) }
+      end
+    end
+  end
+
+  describe 'POST create' do
+    before { post :create, params: params }
+
+    context 'when Success' do
+      context 'with valid name' do
+        let(:params) { { project_id: project.id, name: FFaker::Book.title } }
+
+        it { expect(response).to have_http_status(:created) }
+        it { expect(response).to match_response_schema('task') }
+      end
+    end
+
+    context 'when Failure' do
+      context 'with invalid name' do
+        let(:params) { { project_id: project.id, name: nil } }
```

Comment:
You can use this approach
```
let(:params) do
 project_id: project_id,
 name: name 
end 
```

And change only needed let
```
        let(:name) { nil }
```
Instead of 
```
        let(:params) { { project_id: project.id, name: nil } }
```